components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java

```diff
@@ -126,42 +130,58 @@ public static synchronized void init() {
                 }
             }
 
-            usernameHeaderName = getPropertyValue(Constants.TenantConstants.USERNAME_HEADER);
-            carbonLogin = getPropertyValue(Constants.TenantConstants.USERNAME);
-
-            // Base64 encoded username
-            carbonLogin = Base64.encode(carbonLogin.getBytes(Constants.TenantConstants.CHARACTER_ENCODING));
+            //Initialize the keystores in EndpointConfig.properties only if the "mutualSSLManagerEnabled"
+            // feature is enabled.
+            if (isMutualSSLManagerEnabled(getPropertyValue(Constants.TenantConstants.MUTUAL_SSL_MANAGER_ENABLED))) {
+                usernameHeaderName = getPropertyValue(Constants.TenantConstants.USERNAME_HEADER);
+                carbonLogin = getPropertyValue(Constants.TenantConstants.USERNAME);
 
-            String clientKeyStorePath = buildFilePath(getPropertyValue(Constants.TenantConstants.CLIENT_KEY_STORE));
-            String clientTrustStorePath = buildFilePath(getPropertyValue(Constants.TenantConstants
-                    .CLIENT_TRUST_STORE));
+                // Base64 encoded username
+                carbonLogin = Base64.encode(carbonLogin.getBytes(Constants.TenantConstants.CHARACTER_ENCODING));
 
-            if (StringUtils.isNotEmpty(getPropertyValue(Constants.TenantConstants.TLS_PROTOCOL))) {
-                TenantMgtAdminServiceClient.setProtocol(getPropertyValue(Constants.TenantConstants
-                        .TLS_PROTOCOL));
-            }
+                String clientKeyStorePath = buildFilePath(getPropertyValue(Constants.TenantConstants.CLIENT_KEY_STORE));
+                String clientTrustStorePath = buildFilePath(getPropertyValue(Constants.TenantConstants
+                        .CLIENT_TRUST_STORE));
 
-            if (StringUtils.isNotEmpty(getPropertyValue(Constants.TenantConstants.KEY_MANAGER_TYPE))) {
-                TenantMgtAdminServiceClient.setKeyManagerType(getPropertyValue(Constants.TenantConstants
-                        .KEY_MANAGER_TYPE));
-            }
-            if (StringUtils.isNotEmpty(getPropertyValue(Constants.TenantConstants.TRUST_MANAGER_TYPE))) {
-                TenantMgtAdminServiceClient.setTrustManagerType(getPropertyValue(Constants.TenantConstants
-                        .TRUST_MANAGER_TYPE));
-            }
+                if (StringUtils.isNotEmpty(getPropertyValue(Constants.TenantConstants.TLS_PROTOCOL))) {
```

Comment:
better to check with StringUtils.isNotBlank as these are properties

Reply:
Fixed - https://github.com/wso2/carbon-identity-framework/pull/2559